src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -106,11 +123,13 @@ class LatticeConfigurationDialog extends React.Component {
         );
 
         // preserve basis if asked to do so (eg. when constructing a slab)
-        const newMaterial = new Material(newMaterialConfig);
+        let newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
         newMaterial.isNonPeriodic = this.state.isNonPeriodic;
-
+        if (this.state.isNonPeriodic) {
+            newMaterial = this.handleNonPeriodicLatticeUpdate(newMaterial)
+        }
```

Comment:
There should be no lattice updates for non-periodic systems at all. I don't believe changes to this function are needed.